### ISSUE TITLE
File resources are never released after a thread is interrupted when using asyncWriteEnable()

### ISSUE THREAD

**Issue by fasterdude7** [jankotek/mapdb#490]
Hi this is similar to the previouse issue i fixed:
https://github.com/jankotek/MapDB/issues/335

The problem is nothing to do with writing to a file with a thread that has been interrrupted.
The problem here is the thread gets interrupted but because the thread can not write to the disk it throws an exception and we never get to close the files we opened. This is really painful on windows as we can't delete those files until the JVM shuts down.

I think the problem is in AsyncWriteEngine#close()
the code looks like this.
```
@Override
    public void close() {
        commitLock.writeLock().lock();
        try {
            if(closeInProgress) return;
            checkState();
            closeInProgress = true;
            //...
            //Stuff to ready the actual close and a call to close here
            //...
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }finally {
            commitLock.writeLock().unlock();
        }
    }
```
I think the problem is in the checkState() call, where it throws an exception if the async writter thread caught an exception.

i guess the solution would be to surround the checkState(); call in a try finally block.

**Comment by jankotek** [comment]
Closing. This is already merged into 1.0 and 2.0 was rewritten.

**Comment by fasterdude7** [comment]
It looks like this issue was fixed then the fixed put in at b3a000e and later removed with 34da701
looking at 
https://github.com/jankotek/mapdb/blob/release-1.0/src/main/java/org/mapdb/AsyncWriteEngine.java

It seems that we don't call `AsyncWriteEngine.super.close();` when `checkState();` throws an exception.
```
            if(closeInProgress) return;
            checkState();
            closeInProgress = true;
            try {
                //notify background threads
                if (!action.compareAndSet(null, new CountDownLatch(0)))
                    throw new AssertionError();

                //wait for background threads to shutdown

                while (!activeThreadsCount.await(1000, TimeUnit.MILLISECONDS)) {
                    //nothing here
                }

            }finally {
                AsyncWriteEngine.super.close();
            }
```
should become:
```
            if(closeInProgress) return;
            try {
                checkState();
                closeInProgress = true;
            
                //notify background threads
                if (!action.compareAndSet(null, new CountDownLatch(0)))
                    throw new AssertionError();

                //wait for background threads to shutdown

                while (!activeThreadsCount.await(1000, TimeUnit.MILLISECONDS)) {
                    //nothing here
                }

            }finally {
                AsyncWriteEngine.super.close();
            }
```
